crates/sui-json-rpc-types/src/lib.rs

```diff
@@ -1439,13 +1439,12 @@ pub struct SuiMovePackage {
     pub disassembled: BTreeMap<String, Value>,
 }
 
-impl TryFrom<MoveModulePublish> for SuiMovePackage {
-    type Error = anyhow::Error;
-
-    fn try_from(m: MoveModulePublish) -> Result<Self, Self::Error> {
-        Ok(Self {
-            disassembled: disassemble_modules(m.modules.iter())?,
-        })
+impl From<MoveModulePublish> for SuiMovePackage {
+    fn from(m: MoveModulePublish) -> Self {
+        Self {
+            // In case of failed publish transaction, disassemble can fail, we can only return empty module map in that case.
+            disassembled: disassemble_modules(m.modules.iter()).unwrap_or_default(),
```

Comment:
Might be cleaner to have an option here instead, i.e. `pub disassembled: Option<BTreeMap<String, Value>>,`

Reply:
I will create a follow up PR for this, changing this to Option will be a schema breaking changes.